sdk/servicebus/service-bus/test/streamingReceiverSessions.spec.ts

```diff
@@ -1199,3 +1201,65 @@ describe("Sessions Streaming - Not receive messages after receiver is closed", f
     await testReceiveMessages();
   });
 });
+
+describe("Sessions streaming - close() respects an in progress init()", () => {
+  afterEach(() => {
+    return afterEachTest();
+  });
+
+  it("close() while init() is happening", async () => {
+    sbClient = getServiceBusClient();
+
+    const { receiverClient, senderClient } = await getSenderReceiverClients(
+      sbClient,
+      TestClientType.UnpartitionedQueue,
+      TestClientType.UnpartitionedQueue
+    );
+
+    // force the link open - doesn't affect our tests below but it does
+    // let me mock out the proper method that I need to prove things are working
+    await senderClient.createSender().send({
+      body: "hello",
+      sessionId: TestMessage.sessionId
+    });
+
+    const receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete, {
+      sessionId: TestMessage.sessionId
+    });
+
+    const origReceiverFn = sbClient["_context"].connection["createReceiver"];
+
+    let closePromise: Promise<void> | undefined;
+    let createdReceiver: RheaReceiver | undefined;
+
+    sbClient["_context"].connection["createReceiver"] = async (options) => {
+      // close() will be blocked at this point
+      closePromise = receiver.close();
+
+      const result = await Promise.race([
+        closePromise,
+        delay(2000, "delay_should_win_because_close_is_blocked")
+      ]);
+
+      if (result !== "delay_should_win_because_close_is_blocked") {
+        throw new Error(
+          "FATAL ERROR - test is incorrect because close() was supposed to be blocked but somehow completed"
+        );
+      }
+
+      createdReceiver = await origReceiverFn.apply(sbClient["_context"].connection, [options]);
+      return createdReceiver;
+    };
+
+    // the sessionreceiver uses the same internal (for initialization) as `receiveMessages`
+    // does so I'm just using that for this test since it doesn't require us to coordinate connection
+    // start, etc...
+    await receiver.receiveMessages(1).catch(() => {});
```

Comment:
Once the `init` is done, will we receive the message before the `close`?

Reply:
Oof, you spotted a bug in my test (big time). Fixing...

Reply:
Okay, fix is pushed. The issue you pointed out made me wonder if we even needed to catch an error in the first place (in the non-session based batching receiver closing the connection just results in an empty receive).

Once I removed it I realized that I'd actually chosen the wrong queue (non-partitioned queue!) for this test and then that made me follow through with the rest. 

So to answer your question:
1. It'll probably always throw an error...
2. ...but just in case the JS scheduler isn't always the same I don't make the error being thrown is required since the important bit is the receiver properly being closed at the end.